### PR TITLE
[MRG+1] DOC: Add missing API change docs for 1.5.3

### LINKED ISSUE #7218: v1.5.3: marker=None no longer works in plot()
I just upgraded to v1.5.3 using Anaconda in Python 3.5.2 and noticed that using `marker=None` in `matplotlib.pyplot.plot()` no longer has an effect. Compare:

__v1.5.1__
screen shot 2016-10-04 at 14 04 07

__v1.5.3__
screen shot 2016-10-04 at 14 05 16

### PR DESCRIPTION
**Pull request by tacaswell**
closes #7218

### CODE REVIEW

File: doc/api/api_changes.rst [hunk 1]
```diff
@@ -11,6 +11,26 @@ sources of the changes you are experiencing.
...
 
+Changes in 1.5.3
+================
+
+Default Behavior Changes
+------------------------
+
+Prior to 1.5.3 kwargs passed to `~matplotlib.Axes.plot` were handled
+in two parts -- user supplied kwargs and default kwargs generated
+`~matplotlib.Axes.plot` (such as the cycled styles).  The two
```
Comment (QuLogic) [inline doc/api/api_changes.rst]: Seems like there's a word missing between this line and the above one.

File: doc/api/api_changes.rst [hunk 2]
```diff
@@ -11,6 +11,26 @@ sources of the changes you are experiencing.
...
+dictionaries are now merged and passed into
+`~matplotlib.lines.Line2D.__init__` all at once instead of applying the
+user supplied kwargs via a bulk update after the artist has been created.
```
Comment (QuLogic) [inline doc/api/api_changes.rst]: "via a bulk update" seems extraneous.

File: doc/api/api_changes.rst [hunk 3]
```diff
@@ -11,6 +11,26 @@ sources of the changes you are experiencing.
...
+
+This causes ``None`` passed as ``plot(..., marker=None)`` to now 'do
+the default' thing rather than to result in no marker.  This is only
+an issue if the user explicitly passes ``None`` and has changed the
```
Comment (QuLogic) [inline doc/api/api_changes.rst]: "the user" seems like a different PoV than the rest of this document, though it's not perfectly consistent anyway...

**Comment by tacaswell** [comment]
@qulogic basicly re-wrote it.

File: doc/api/api_changes.rst [hunk 4]
```diff
@@ -11,6 +11,34 @@ sources of the changes you are experiencing.
...
+================
+
+``ax.plot(..., marker=None)`` gives default marker
+--------------------------------------------------
+
+Prior to 1.5.3 kwargs passed to `~matplotlib.Axes.plot` were handled
+in two parts -- default kwargs generated internal to
+`~matplotlib.Axes.plot` (such as the cycled styles) and user supplied
+kwargs.  The internally generated kwargs were passed to the
+`~matplotlib.lines.Line2D.__init__` and the user kwargs were passed to
```
Comment (QuLogic) [inline doc/api/api_changes.rst]: That would generate as "The internally generated kwargs were passed to the __init__ and ...", right? It seems like using the full name would be a good idea here (no `~`.)

File: doc/api/api_changes.rst [hunk 5]
```diff
@@ -11,6 +11,34 @@ sources of the changes you are experiencing.
...
+is the convention through out mpl rather than raising an exception.
+
+Unlike most `~matplotlib.lines.Line2D` setter methods
+`~matplotlib.lines.Line2D.set_marker` did accept `None` as a valid
+input which was mapped to 'no marker'.  Thus, by routing this
+``marker=None`` through ``__init__`` rather than ``set(...)`` the meaning
+of ``ax.plot(..., marker=None)`` changed from 'no markers' to 'default markers
+from rcparams'.
+
+This is change in only evident if ``mpl.rcParams['lines.marker']`` has a value
```
Comment (QuLogic) [inline doc/api/api_changes.rst]: "This is change in" -> "This change is", I think.

**Comment by NelleV** [comment]
Thanks!

**Pull request merged**